src/classes/RD2_EntryFormController.cls

```diff
@@ -318,17 +344,12 @@ public with sharing class RD2_EntryFormController {
 
         /**
         * @description Handles validation and Elevate recurring commitment creation
-        * @param jsonRecord Recurring Donation fields as entered by the user before the record is saved
+        * @param rd Recurring Donation
         * @param paymentMethodToken Payment Method Token
         * @return UTIL_Http.Response Payments API response
         */
-        public UTIL_Http.Response handleCommitment(String jsonRecord, String paymentMethodToken) {
+        public UTIL_Http.Response handleCommitment(npe03__Recurring_Donation__c rd, String paymentMethodToken) {
```

Comment:
extract rd JSON deserialize and validation out so the method only serves the purpose of handling commitment API